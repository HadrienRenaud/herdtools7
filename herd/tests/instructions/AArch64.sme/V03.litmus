AArch64 V03
{
  uint32_t x[4] = {1,2,3,4};
  0:X0=x;
}
P0 ;
SMSTART                       ;
PTRUE P0.S,VL4                ;
MOV W12,WZR                   ;
LD1W {ZA1H.S[W12,0]},P0/Z,[X0];
MOVA Z0.S,P0/M,ZA1H.S[W12,0]  ;
UADDV D1,P0,Z0.S              ;
FMOV W1,S1                    ; (* Stash result in GP register or SMSTOP would zero it out *)
SMSTOP                        ;
forall 0:X1=10
