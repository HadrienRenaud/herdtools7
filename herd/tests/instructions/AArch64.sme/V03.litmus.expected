Test V03 Required
States 1
0:X1=10;
Ok
Witnesses
Positive: 1 Negative: 0
Flag Scalable-Matrix-Extensions-is-work-in-progress
Condition forall (0:X1=10)
Observation V03 Always 1 0
Hash=3735f297bb64925ffd8f875869f0d2b1

